sdk/src/main/java/com/mapbox/maps/renderer/widget/Widget.kt

```diff
@@ -10,12 +10,44 @@ abstract class Widget internal constructor() {
   internal abstract val renderer: WidgetRenderer
 
   /**
-   * Set absolute translation of widget in pixels.
+   * Update the widget to the new position.
    */
-  abstract fun setTranslation(translateX: Float, translateY: Float)
+  abstract fun setPosition(widgetPosition: WidgetPosition)
 
   /**
-   * Set absolute rotation of widget in angles.
+   * Get the current position of the widget.
+   */
+  abstract fun getPosition(): WidgetPosition
+
+  /**
+   * Set the translation of the widget in pixels, relative to it's current position.
+   *
+   * @param translateX the offset in pixels towards the right of the screen.
+   * @param translateY the offset in pixels towards the bottom of the screen.
+   */
+  @Deprecated(
+    message = "setTranslation is deprecated, please use setPosition instead.",
+    replaceWith = ReplaceWith("setPosition")
+  )
+  fun setTranslation(translateX: Float, translateY: Float) {
+    val currentPosition = getPosition()
+    setPosition(
+      WidgetPosition {
+        horizontalAlignment = currentPosition.horizontalAlignment
+        verticalAlignment = currentPosition.verticalAlignment
+        offsetX = currentPosition.offsetX + translateX
+        offsetY = currentPosition.offsetY + translateY
+      }
+    )
+  }
+
+  /**
+   * Get the absolute rotation of widget in degrees.
    */
   abstract fun setRotation(angleDegrees: Float)
+
+  /**
+   * Set absolute rotation of widget in degrees.
```

Comment:
This comment belongs to `setRotation` and the one in `setRotation` belongs there.

Reply:
good catch, fixed.